django/db/models/fields/reverse_related.py

```diff
@@ -138,6 +138,16 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.identity)
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        # We delete the cached_property here because it can be re-calculated at
+        # first invocation after loads() is used.
+        # We need to remove it for ubclasses like ManyToOneRel where the
+        # PathInfo may contain an intermediate M2M table that's been
+        # dynamically created and doesn't exist in the .models module.
+        state.pop('path_infos', None)
+        return state
```

Comment:
This is implemented because otherwise deep down in a query's `alias_map` a `Join` can include a rel one side of which is an internal m2m intermediate table (`<class 'queryset_pickle.models.M2MModel_groups'>`). Tested by `tests.queryset_pickle.tests.PickleabilityTestCase.test_filter_reverse_fk` and `...test_pickle_filteredrelation`